File: Algorithm/c_or_cpp/charpter1_linklist/level2/topic2_5删除/remove_elements.cpp
#include <stdlib.h>
#include <stdio.h>

#include <stdio.h>
#include "../../list_node.cpp"

struct ListNode* removeElements(struct ListNode* head, int val) {
    struct ListNode* dummyHead = (struct ListNode*) malloc(sizeof(struct ListNode));
    dummyHead->val = 0;
    dummyHead->next = head;
    struct ListNode* cur = dummyHead;
    while (cur->next != NULL) {
        if (cur->next->val == val) {
            struct ListNode* temp = cur->next;
            cur->next = cur->next->next;
            free(temp);
        } else {
            cur = cur->next;
        }
    }
    struct ListNode* newHead = dummyHead->next;
    free(dummyHead);
    return newHead;
}

int main()
{
    struct ListNode *list1 = NULL;

    int a[] = {1, 2, 6, 3, 4, 5, 6};

    list1 = initLink(a, 7);

   removeElements(list1, 6);

    printList(list1);
    return 0;
}

File: Algorithm/Code_Materials/python_source/charpter1_linklist/level2/topic2_5删除/remove_elements.py
# coding=utf-8
from python_source.list_node import init_list


class RemoveElements:
    def removeElements(self, head, val):
        while head and head.val == val:
            head = head.next
        if head is None:
            return head
        node = head
        while node.next:
            if node.next.val == val:
                node.next = node.next.next
            else:
                node = node.next
        return head


if __name__ == '__main__':
    array = [1, 2, 6, 3, 4, 5, 6]
    val = 6
    head = init_list(array)
    removeElements = RemoveElements()
    node = removeElements.removeElements(head, 6)
    print  node
